### PR TITLE
Sort

### LINKED ISSUE #9852: sort

### PR DESCRIPTION
**Pull request by yumnaAlshalak**
Close #9852

### CODE REVIEW

**Comment by yumnaAlshalak** [comment]
What is the wrong of my code?
Is this because of spaces or there is something else?

**Comment by yumnaAlshalak** [comment]
Close #9852

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Hey! 
Thank you soo much for contributing to ivy! 🙂
Your PR looks amazing ✨!

Actually there is one thing which you forgot about maybe, you have not added the test for the specific function. I'll link you to the guide which will help you explain in detail on how to add the test function. 
https://lets-unify.ai/ivy/deep_dive/ivy_frontends_tests.html

Thanks! 🙂

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Awesome, thanks for the changes! 🙂
Just a few more changes required before we can merge your PR, I've mentioned them below!

You've mistakenly added both of the files on the main repo, they should rather be in their respective directories 🙂 

1. For your functional implementation it should be in `ivy/functional/frontends/jax/numpy/searching_sorting.py` 
2. Incase of the implementation of the test, it should be here, `ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py`

Thanks! 🙂

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Your test is failing, there are some minor changes which need to be done.
There was a refactor in the frontend tests recently, you have to use `test_flags` now which handle multiple arguments.

I'll link you to an example which would help you in the refactor. https://github.com/unifyai/ivy/blob/master/ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py#L102

Thanks! 🙂

File: ivy_tests/test_ivy/test_frontends/test_jax/sort_test.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        min_axis=-1,
+        max_axis=0,
+        min_num_dims=1,
+        force_int_axis=True,
+    ),
+)
+def test_numpy_sort(
+    *,
+    dtype_x_axis,
+    as_variable,
```
Comment (zaeemansari70) [inline ivy_tests/test_ivy/test_frontends/test_jax/sort_test.py:18]: this and some below

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
You forgot to remove the arguments which are not being used after the refactor, such as `as_variable`.
Take a little closer look at this example again, you'll know 🙂
https://github.com/unifyai/ivy/blob/master/ivy_tests/test_ivy/test_frontends/test_torch/test_linalg.py#L102 
Thanks!

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
You have made separate files for the functions, this should not be the case,
Some changes are requested below:

1. Your frontend function should be here, ivy/functional/frontends/jax/numpy/searching_sorting.py
2. Your test function should be here, ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_searching_sorting.py
 
Also please update your branch so the new changes can be reflected from ivy upstream to your branch. Your code is giving errors while testing locally, as your branch is out of date.

 Do let me know if you have any questions.
 Thanks! 🙂

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Hey @yumnaAlshalak ,
Thanks for the changes!
The `intelligent-tests-pr / run_tests_1 (pull_request)` are failing in the logs and showing import errors, kindly fix those so we can merge your PR. 
https://github.com/unifyai/ivy/actions/runs/4078620822/jobs/7029061320
Do let me know if you have any other questions!
Thanks! 🙂

**Comment by yumnaAlshalak** [comment]
Hi sir @zaeemansari70 
How can I know what is the error ?
The link didn't show me what it is
It showed me only this:
 shell: /usr/bin/bash -e {0}

**Comment by yumnaAlshalak** [comment]
Hi sir @zaeemansari70 
I tried to understand the new error, but I couldn't.
Can you explain it to me to solve it?

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Hi @yumnaAlshalak , actually you are having an assertion error which says that:
`E   AssertionError: [-0.5 -1. ] != [-1.  -0.5]`,  here https://github.com/unifyai/ivy/actions/runs/4119031904/jobs/7112253753 if you expand the `Determine and Run Tests` window.

This actually means that the value which is being produced by the ground_truth_frontend is not matching the value produced by your implementation.
The ground truth is the original function of jax, they should match to make the test pass.
There might be a small logical error in your code which you have to debug and fix.

Do let me know if you have any other questions!
Thanks! 🙂 

P.S Please don't call me "Sir" 😅

**Comment by yumnaAlshalak** [comment]
Hi @zaeemansari70 
I think there is an error with libraries.
Why did this happen? I didn't change anything.

File: ivy/functional/frontends/jax/numpy/searching_sorting.py [hunk 2]
```diff
@@ -90,3 +98,26 @@ def extract(condition, arr):
...
+def sort(
+    x: Union[ivy.Array, ivy.NativeArray],
+    /,
+    *,
+    axis: int = -1,
+    descending: bool = False,
+    stable: bool = True,
+    out: Optional[ivy.Array] = None,
+):
+    if axis == 1:
```
Comment (zaeemansari70) [inline ivy/functional/frontends/jax/numpy/searching_sorting.py:117]: kindly debug your code from here as the difference is very minimal as the assertion error is `[1 0] != [0 1]`, with the original framework, you are missing some logical implementation while implementing this

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
I guess you forgot to request a review from me, I have reviewed your PR again and requested some changes!
Do request a review from me when you are done with your changes!
Also let me know if you have any other questions!
Thanks! 🙂

File: ivy/functional/frontends/jax/numpy/searching_sorting.py (hunk 2 above)
Comment (yumnaAlshalak) [reply ivy/functional/frontends/jax/numpy/searching_sorting.py:117]: No, I didn't forget but I am still trying to solve the problem every day.

**Review by zaeemansari70 — APPROVED** [review]
Lgtm!

**Comment by yumnaAlshalak** [comment]
What's the next step? @zaeemansari70

**Comment by zaeemansari70** [comment]
Hi @yumnaAlshalak , maybe text in the hiring channel on discord tagging the hiring team, they would have a better idea, I hope it helps :slightly_smiling_face:

**Pull request merged**